APMrover2/crash_check.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Rover.h"
+
+// Code to detect a crash or block
+#define CRASH_CHECK_TRIGGER_SEC     2       // 2 seconds blocked indicates a crash
+#define CRASH_CHECK_THROTTLE_MIN    5.0f    // vehicle must have a throttle greater that 5% to be considered crashed
+#define CRASH_CHECK_VEL_MIN         0.08f   // vehicle must have a velocity under 0.08 m/s to be considered crashed
+
+// crash_check - disarms motors if a crash or block has been detected
+// crashes are detected by the vehicle being static (no speed) for more than CRASH_CHECK_TRIGGER_SEC and motor are running
+// called at 10Hz
+void Rover::crash_check()
+{
+  static uint16_t crash_counter;  // number of iterations vehicle may have been crashed
+
+  // return immediately if disarmed, or crash checking disabled or in HOLD mode
+  if (!arming.is_armed() || g.fs_crash_check == 0 || (control_mode != GUIDED && control_mode != AUTO)) {
```

Comment:
@muramura A global defines.h will be very difficult because each vehicle has their own definitions. But if you want to try and do it, I would put in libraries/AP_Common